hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -2821,7 +2822,23 @@ else if ( isAllOrDirtyOptLocking() && oldFields != null ) {
 					String[] propertyColumnWriters = getPropertyColumnWriters( i );
 					boolean[] propertyNullness = types[i].toColumnNullness( oldFields[i], getFactory() );
 					for ( int k = 0; k < propertyNullness.length; k++ ) {
-						if ( propertyNullness[k] ) {
+						if ( propertyColumnNames[k] == null && propertyColumnFormulaTemplates[k] != null ) {
+							String formula = StringHelper.replace(
+									propertyColumnFormulaTemplates[k][i],
```

Comment:
Shouldn't this be the other way around? `i` is the property index and `k` seems to be the column span index for the current property.

```suggestion
									propertyColumnFormulaTemplates[i][k],
```